app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -2808,16 +2855,34 @@ private static boolean hasContentFilter(Set<SearchFilter> filters) {
         return false;
     }
 
-    protected static long nextContentId(Handle handle) {
-        return handle.createQuery("SELECT nextval('contentidsequence')")
-                .mapTo(Long.class)
-                .one();
+    protected long nextContentId(Handle handle) {
+        return nextSequenceValue(handle, CONTENT_ID_SEQUENCE);
     }
 
-    protected static long nextGlobalId(Handle handle) {
-        return handle.createQuery("SELECT nextval('globalidsequence')")
-                .mapTo(Long.class)
-                .one();
+    protected long nextGlobalId(Handle handle) {
+        return nextSequenceValue(handle, GLOBAL_ID_SEQUENCE);
+    }
+
+    private long nextSequenceValue(Handle handle, String sequenceName) {
+        String sql = sqlStatements.getNextSequenceValue();
+        if ("postgresql".equals(sqlStatements.dbType())) {
+            return handle.createQuery(sql)
+                    .bind(0, tenantContext.tenantId())
+                    .bind(1, sequenceName)
+                    .mapTo(Long.class)
+                    .one();
+        } else {
+            handle.createUpdate(sql)
```

Comment:
If we did SELECT in the first place, we could add FOR UPDATE, that would lock the selected rows to be effected until transaction ends. This could solve the concurrent problem.
The only problem that will stay is when there is no row for the sequence, so it won't be selected and won't be locked. This could be solved by "creating" sequences when creating the tenants.